src/diffusers/loaders/single_file_utils.py

```diff
@@ -442,6 +330,76 @@ def infer_model_type(original_config, checkpoint, model_type=None):
     return model_type
 
 
+def is_clip_model(checkpoint):
+    if CHECKPOINT_KEY_NAMES["clip"] in checkpoint:
+        return True
+
+    return False
+
+
+def is_open_clip_model(checkpoint):
+    if CHECKPOINT_KEY_NAMES["open_clip"] in checkpoint:
+        return True
+
+    return False
+
+
+def is_open_clip_sdxl_model(checkpoint):
+    if CHECKPOINT_KEY_NAMES["open_clip_sdxl"] in checkpoint:
+        return True
+
+    return False
+
+
+def is_clip_model_in_single_file(class_obj, checkpoint):
+    is_clip_in_checkpoint = any(
+        [is_clip_model(checkpoint), is_open_clip_model(checkpoint), is_open_clip_sdxl_model(checkpoint)]
+    )
+    if (
+        class_obj.__name__ == "CLIPTextModel" or class_obj.__name__ == "CLIPTextModelWithProjection"
+    ) and is_clip_in_checkpoint:
+        return True
+
+    return False
+
+
+def fetch_diffusers_config(checkpoint, subfolder=None):
```

Comment:
Just so I'm understand correctly, the suggestion for `original_config` is to ignore using it to configure the model and just use it to fetch the diffusers hub config? 

I'm cool with this. The only potential issue is if there are original configs that have changed parameters (this is quite unlikely I think). But if the parameters are changed, I think it could be reasonable to just set them by passing them in as kwargs to `from_single_file`.  IMO this is more "correct" because it follows how it's done in `from_pretrained`

Reply:
> It is a bit breaking, but we could raise a warning saying something like "original_config is only used to determine the pipeline config to use. If you are trying to customize the configuration the model class, please pass these arguments directly to from_single_file

Could you show a pseudo code for that?

Reply:
>Just so I'm understand correctly, the suggestion for original_config is to ignore using it to configure the model and just use it to fetch the diffusers hub config?

I think yes!!  I'm not sure how we should use it to fetch diffusers hub config, though -  what information can the config file provide that we cannot infer from the checkpoint? 

I looked at the code again, I think currently, in this PR, `original_config` is only used to find out `image_size` (is it the case? I could be wrong....), which should be the same  as the `sample_size` we could get directly from the hub config, and we would not need the `original_config` file to fetch the hub config 


>It is a bit breaking, but we could raise a warning saying something like "original_config is only used to determine the pipeline config to use. If you are trying to customize the configuration the model class, please pass these arguments directly to from_single_file

sounds good!

Reply:
>Just so I'm understand correctly, the suggestion for original_config is to ignore using it to configure the model and just use it to fetch the diffusers hub config?

ohh actually, there might be a misunderstanding here - I'm only referring to the `infer_model_type` function that you use to infer the model type based on `original_config`; we can move it up to inside `fetch_diffusers_config`, or just remove the function if it does not provide additional info

Reply:
>the suggestion for original_config is to ignore using it to configure the model and just use it to fetch the diffusers hub config?

and for this, even though it is not my original intention, I'm cool with it, too, if it is unlikely to have models with changed configurations! I think it might indeed be unnecessary with this PR refractory 

I'd like to cc @vladmandic here for this opinion: Will there still be a need to pass a' original_config`, now that we can map the checkpoint to the corresponding hub config now?

Reply:
> this is great, just lets ensure that configs location can be a local folder as well as huggingface.co itself as there are plenty of users without internet connectivity (corp firewalls, great wall of china, etc.)
(in that case, it would be up to app to have all config files available offline)

I think the main use case for original config that @vladmandic is referring to. Having a single local config file for the checkpoint instead of fetching the multiple ones for the individual models from the hub.

Reply:
> I think yes!! I'm not sure how we should use it to fetch diffusers hub config, though - what information can the config file provide that we cannot infer from the checkpoint?

Scheduler type and its configuration (off the top of my head)? 

But at the same time, in case of `local_files_only=True` we're limited to the information provided by the users. In that case, we cannot query the Hub and may have to rely on the config file being provided by the user, instead. 

If we can clearly distinguish the things we need the original config file for, that will help clear a lot of these questions, in my opinion.

Reply:
> I'd like to cc @vladmandic here for this opinion: Will there still be a need to pass a' original_config`, now that we can map the checkpoint to the corresponding hub config now?

Absolutely - one of key requirements is that models must work offline without internet connectivity - that is the main reason for having model configs locally and passing them as original_config. Its quite common that server->huggingface is not allowed from corp network (it may be allowed from user's workstation, but not the server attempting to load a model).

I'm ok with any option that allows to meet strict-offline requirement, I'm not going to prescribe "how".
One option would be to simply have option to set config path prefix in loadermixin so whatever model config is autodetermined, it would be loaded from local path or url instead of huggingface.co.
Also, I'm totally ok with using either SAI-style YAML configs or Diffusers-style JSON configs.

Reply:
@vladmandic I would prefer to use the Diffusers style JSON configs as it's easier to configure stuff exactly as expected in the model object, and there's a lot of utils we can reuse. 

The configs might need to live in nested directories? Does that work? e.g A single file model would look something like this

```
.
├── model_index.json
├── playground-v2.5-1024px-aesthetic.fp16.safetensors
├── scheduler
│   └── scheduler_config.json
├── text_encoder
│   └── config.json
├── text_encoder_2
│   └── config.json
├── unet
│   └── config.json
└── vae
    └── config.json
```

Reply:
@DN6 i'm ok with json configs, but maybe we can simplify it - i really don't see a point of a 5+ config files when loading a single-file.

Reply:
@vladmandic Hmm I think trying to create a new JSON config type might be overkill for this PR. Will work to make this backwards compatible and we can revisit some config options in a future PR.

Reply:
ok, but how do you see offline configs would be specified/used then? and how can i pre-populate them?